### PR TITLE
Rdm 4575 search filters definition service

### PR DESCRIPTION
**Pull request by adnanakgun**
### JIRA link (if applicable) ###

https://tools.hmcts.net/jira/browse/RDM-4575

### Change description ###

Extracted definitionsService from man-web and implemented it here. Modified SearchFiltersComponent to consume the service. Tweaked tests.

**Does this PR introduce a breaking change?** (check one with "x")

```
[x] Yes
[ ] No
```

### CODE REVIEW

**Comment by adnanakgun** [comment]
@andrewfolga do check amendments please.

File: src/shared/components/search-filters/search-filters-wrapper.component.ts [hunk 1]
```diff
@@ -0,0 +1,63 @@
+import { Component, OnInit, Input, Output, EventEmitter } from '@angular/core';
+import { Jurisdiction } from '../../domain';
+import { DefinitionsService } from '../../services';
+import { READ_ACCESS } from '../../domain/case-view/access-types.model';
+
+@Component({
+    selector: 'ccd-search-filters-wrapper',
+    template: `
+        <ccd-search-filters
```
Comment (andrewfolga) [inline src/shared/components/search-filters/search-filters-wrapper.component.ts]: Can we have a template in a separate file to be consistent?

File: demo/src/app/search-filters-consumer.component.ts [hunk 2]
```diff
@@ -8,14 +8,58 @@ import { Jurisdiction, CaseType } from '@hmcts/ccd-case-ui-toolkit';
 })
 export class SearchFiltersConsumerComponent implements OnInit {
```
Comment (andrewfolga) [inline demo/src/app/search-filters-consumer.component.ts]: We do not need this one in demo app if we have a wrapper. Please remove.

File: demo/src/app/search-filters-consumer.component.html [hunk 3]
```diff
@@ -28,7 +30,7 @@ <h1 class="heading-large">Search Filters component</h1>
                                     </ul>
                         </li>
                         <li>onReset - a notification event about a reset button being clicked</li>
-                        <li>onJurisdiction - a notification event with selected jurisdiction object</li>
+                        <li>onJuridiction - a notification event with selected jurisdiction object</li>
```
Comment (andrewfolga) [inline demo/src/app/search-filters-consumer.component.html]: fix typo

File: demo/src/app/search-filters-consumer.component.html [hunk 4]
```diff
@@ -3,10 +3,12 @@
         <h1 class="heading-large">Search Filters component</h1>
```
Comment (andrewfolga) [inline demo/src/app/search-filters-consumer.component.html]: We do not need this one in demo app if we have a wrapper. Please remove.

File: demo/src/app/app.routing.ts [hunk 5]
```diff
@@ -31,6 +32,9 @@ export const routing = RouterModule.forRoot([
       { path: 'search',
```
Comment (andrewfolga) [inline demo/src/app/app.routing.ts]: We do not need this one in demo app if we have a wrapper. Please remove.

File: demo/src/app/app.routing.ts [hunk 6]
```diff
@@ -7,6 +7,7 @@ import { CaseViewConsumerComponent } from './case-view-consumer.component';
 import { SearchFiltersConsumerComponent } from './search-filters-consumer.component';
```
Comment (andrewfolga) [inline demo/src/app/app.routing.ts]: We do not need this one in demo app if we have a wrapper. Please remove.

File: demo/src/app/core.component.html [hunk 7]
```diff
@@ -13,6 +13,9 @@
             <li>
                 <a [routerLink]="'/search'" [routerLinkActive]="'item-bold'">Search</a>
```
Comment (andrewfolga) [inline demo/src/app/core.component.html]: We do not need this one in demo app if we have a wrapper. Please remove.

File: demo/src/app/app.module.ts [hunk 8]
```diff
@@ -46,6 +49,7 @@ import { CaseTimelineConsumerComponent } from './case-timeline-consumer.componen
     CaseProgressConsumerComponent,
     CaseViewConsumerComponent,
     SearchFiltersConsumerComponent,
```
Comment (andrewfolga) [inline demo/src/app/app.module.ts]: We do not need this one in demo app if we have a wrapper. Please remove.

File: demo/src/app/app.module.ts [hunk 9]
```diff
@@ -38,6 +40,7 @@ import { CaseTimelineConsumerComponent } from './case-timeline-consumer.componen
     CreateCaseFiltersModule,
     CaseTimelineModule,
     routing,
+    SearchFiltersModule,
```
Comment (andrewfolga) [inline demo/src/app/app.module.ts]: 👍 as it was missing here

File: src/case-ui-toolkit.module.ts [hunk 10]
```diff
@@ -72,7 +72,6 @@ import { CaseHistoryModule } from './shared/components/case-history';
       CaseProgressComponent,
       CaseViewComponent,
       CaseViewerComponent,
-      SearchFiltersComponent,
```
Comment (andrewfolga) [inline src/case-ui-toolkit.module.ts]: 👍

File: src/shared/components/search-filters/search-filters-wrapper.component.ts (hunk 1 above)
Comment (adnanakgun) [reply src/shared/components/search-filters/search-filters-wrapper.component.ts]: Done.

File: demo/src/app/search-filters-consumer.component.ts (hunk 2 above)
Comment (adnanakgun) [reply demo/src/app/search-filters-consumer.component.ts]: Done.

**Comment by adnanakgun** [comment]
Please note I changed a file name and the reference to it:
`search-filters.html` => `search-filters.component.html`

**Pull request merged**